File: ru.kmorozov.library.data.model/src/main/java/ru/kmorozov/library/data/config/MongoConfiguration.java
package ru.kmorozov.library.data.config;

import com.mongodb.MongoClient;
import com.mongodb.MongoClientURI;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.mongodb.config.AbstractMongoConfiguration;

import java.util.Collection;
import java.util.Collections;

/**
 * Created by sbt-morozov-kv on 28.11.2016.
 */

@Configuration
public class MongoConfiguration extends AbstractMongoConfiguration {

    public static final String BOOKS_MONGO_DB_NAME = "BOOKS";

    @Value("${mongo.uri}")
    private String mongoUri;

    @Override
    protected String getDatabaseName() {
        return MongoConfiguration.BOOKS_MONGO_DB_NAME;
    }

    @Override
    public MongoClient mongoClient() {
        return new MongoClient(new MongoClientURI(this.mongoUri));
    }

    @Override
    protected Collection<String> getMappingBasePackages() {
        return Collections.singleton("ru.kmorozov.library.data.model");
    }
}

File: ru.kmorozov.library.data.model/src/main/java/ru/kmorozov/library/data/config/MongoConfiguration.kt
package ru.kmorozov.library.data.config

import com.mongodb.MongoClient
import com.mongodb.MongoClientURI
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Configuration
import org.springframework.data.mongodb.config.AbstractMongoConfiguration
import java.util.Collections

/**
 * Created by sbt-morozov-kv on 28.11.2016.
 */

@Configuration
open class MongoConfiguration : AbstractMongoConfiguration() {

    @Value("\${mongo.uri}")
    private val mongoUri: String? = null

    override fun getDatabaseName(): String {
        return BOOKS_MONGO_DB_NAME
    }

    override fun mongoClient(): MongoClient {
        return MongoClient(MongoClientURI(mongoUri!!))
    }

    override fun getMappingBasePackages(): Collection<String> {
        return setOf("ru.kmorozov.library.data.model")
    }

    companion object {

        const val BOOKS_MONGO_DB_NAME = "BOOKS"
    }
}
